digits/model/tasks/caffe_train.py

```diff
@@ -806,7 +833,33 @@ def task_arguments(self, resources, env):
                     args.append('--gpu=%s' % ','.join(identifiers))
         if self.pretrained_model:
             args.append('--weights=%s' % ','.join(map(lambda x: self.path(x), self.pretrained_model.split(':'))))
+        return args
+
 
+    def _pycaffe_args(self, gpu_id):
+        """
+        Helper to generate pycaffe Python script
+        Returns a list of strings
+
+        Arguments:
+        gpu_id -- the GPU device id to use
+        """
+        # TODO: Remove this once caffe.exe works fine with Python Layer
+        gpu_script = "caffe.set_device({id});".format(id=gpu_id) if gpu_id else ""
```

Comment:
Hey @IsaacYangSLA are you sure this actually runs in GPU mode? Don't you need a `set_mode()` too?